### PR TITLE
F/pwauth

### PR DESCRIPTION
**Pull request by Taliik**

### CODE REVIEW

File: test/CrateTest/PDO/AbstractPDOPwAuthTest.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        $this->pdo->query($priv);
+    }
+
+    protected function tearDown()
+    {
+        $del = "DROP USER IF EXISTS test_user";
+        $this->pdo->query($del);
+    }
+
+
```
Comment (christianbader) [inline test/CrateTest/PDO/AbstractPDOPwAuthTest.php]: remove empty lines

File: test/CrateTest/PDO/PDOPwAuthTest.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * @covers ::<public>
+ *
+ * @group unit
+ */
+class PDOPwAuthTest extends AbstractPDOPwAuthTest
+{
+    public function testAuthentication()
+    {
+
+        $conn = new PDO('crate:localhost:4200', "test_user", "pwd");
```
Comment (christianbader) [inline test/CrateTest/PDO/PDOPwAuthTest.php]: also use single quotes for `test_user` and `pwd`

File: test/provisioning/crate.yml [hunk 3]
```diff
@@ -91,19 +91,19 @@ auth.host_based.enabled: true
...
-      1:
-        user: crate
-        method: trust
-
-      99:
-        method: password
+ host_based:
+   config:
+#     0:
+#       user: crate
```
Comment (christianbader) [inline test/provisioning/crate.yml]: can we remove this?

File: test/CrateTest/PDO/PDOPwAuthTest.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        $conn->setAttribute(PDO::CRATE_ATTR_SSL_MODE, PDO::CRATE_ATTR_SSL_MODE_ENABLED_BUT_WITHOUT_HOST_VERIFICATION);
+
+        $st = $conn->prepare('SELECT CURRENT_USER;');
+
+        self::assertEquals('test_user', $st->fetch()[0]);
+
```
Comment (christianbader) [inline test/CrateTest/PDO/PDOPwAuthTest.php]: remove empty line

File: test/CrateTest/PDO/PDOPwAuthTest.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
```
Comment (christianbader) [inline test/CrateTest/PDO/PDOPwAuthTest.php]: also here

File: test/CrateTest/PDO/AbstractPDOPwAuthTest.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+abstract class AbstractPDOPwAuthTest extends TestCase
+{
+    /**
+     * @var PDO
+     */
+    protected $pdo;
+
+    protected function setUp()
+    {
+        $this->pdo = new PDO('crate:localhost:4200', 'crate', 'crate');
```
Comment (christianbader) [inline test/CrateTest/PDO/AbstractPDOPwAuthTest.php]: Why is this a class variable? Could also be local like: $pdo = new ....

File: test/CrateTest/PDO/AbstractPDOPwAuthTest.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        $this->pdo->setAttribute(PDO::CRATE_ATTR_SSL_MODE, PDO::CRATE_ATTR_SSL_MODE_ENABLED_BUT_WITHOUT_HOST_VERIFICATION);
```
Comment (christianbader) [inline test/CrateTest/PDO/AbstractPDOPwAuthTest.php]: > 120?

File: test/CrateTest/PDO/PDOPwAuthTest.php [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+ */
+class PDOPwAuthTest extends AbstractPDOPwAuthTest
+{
+    public function testAuthentication()
+    {
+
+        $conn = new PDO('crate:localhost:4200', "test_user", "pwd");
+        $conn->setAttribute(PDO::CRATE_ATTR_SSL_MODE, PDO::CRATE_ATTR_SSL_MODE_ENABLED_BUT_WITHOUT_HOST_VERIFICATION);
+
+        $st = $conn->prepare('SELECT CURRENT_USER;');
```
Comment (christianbader) [inline test/CrateTest/PDO/PDOPwAuthTest.php]: can we use a more meaningful variable name here?

File: test/CrateTest/PDO/AbstractPDOPwAuthTest.php [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    protected function tearDown()
+    {
+        $del = "DROP USER IF EXISTS test_user";
+        $this->pdo->query($del);
+    }
+
+
+}
```
Comment (turbo-ele) [inline test/CrateTest/PDO/AbstractPDOPwAuthTest.php]: add empty line at end of file

File: test/CrateTest/PDO/AbstractPDOPwAuthTest.php [hunk 10]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+
```
Comment (turbo-ele) [inline test/CrateTest/PDO/AbstractPDOPwAuthTest.php]: remove empty line

File: test/CrateTest/PDO/PDOPwAuthTest.php [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace CrateTest\PDO;
+
+use Crate\PDO\PDO;
+
+/**
+ * Tests for {@see \Crate\PDO\PDO}
+ *
+ * @coversDefaultClass \Crate\PDO\PDO
+ * @covers ::<public>
```
Comment (turbo-ele) [inline test/CrateTest/PDO/PDOPwAuthTest.php]: is this correct?

File: test/CrateTest/PDO/PDOPwAuthTest.php [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * @group unit
```
Comment (turbo-ele) [inline test/CrateTest/PDO/PDOPwAuthTest.php]: don't know what this is about :)
but do we really need that header here? do we have that somewhere else?

File: test/CrateTest/PDO/PDOPwAuthTest.php [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+ */
+class PDOPwAuthTest extends AbstractPDOPwAuthTest
+{
+    public function testAuthentication()
+    {
+
```
Comment (turbo-ele) [inline test/CrateTest/PDO/PDOPwAuthTest.php]: remove

File: test/CrateTest/PDO/AbstractPDOPwAuthTest.php (hunk 7 above)
Comment (Taliik) [reply test/CrateTest/PDO/AbstractPDOPwAuthTest.php]: no, it's 109

File: test/CrateTest/PDO/AbstractPDOPwAuthTest.php (hunk 6 above)
Comment (christianbader) [reply test/CrateTest/PDO/AbstractPDOPwAuthTest.php]: i see

File: test/CrateTest/PDO/AbstractPDOPwAuthTest.php (hunk 7 above)
Comment (christianbader) [reply test/CrateTest/PDO/AbstractPDOPwAuthTest.php]: ok

File: test/CrateTest/PDO/PDOPwAuthTest.php (hunk 12 above)
Comment (Taliik) [reply test/CrateTest/PDO/PDOPwAuthTest.php]: the header was taken over by the other PDOTest, i don't think it is needed here

**Pull request merged**